mlrun/feature_store/retrieval/job.py

```diff
@@ -13,18 +13,25 @@
 # limitations under the License.
 #
 import uuid
+from typing import Optional
 
 import mlrun
-from mlrun.model import new_task
+from mlrun.model import DataTargetBase, new_task
 from mlrun.runtimes.function_reference import FunctionReference
 from mlrun.utils import logger
 
+from ...runtimes import RuntimeKinds
 from ..common import RunConfig
+from .base import BaseMerger
 
 
 def run_merge_job(
     vector,
-    target,
+    target: DataTargetBase,
+    merger: BaseMerger,
+    engine: str,
+    engine_args: dict,
+    spark_service: Optional[str] = None,
```

Comment:
All of these should basically be `Optional[...] = None` - why putting it just on `spark_service`?